mapusaurus/geo/views.py

```diff
@@ -9,43 +8,38 @@
 from rest_framework.renderers import JSONRenderer
 from rest_framework.response import Response
 from geo.models import Geo
-from respondents.models import Institution
 
-def tract_centroids_as_json(request):
-    return json.loads(tract_centroids_in_json_format(request))
+def geo_as_json(geos):
+    return json.loads(format_geo_to_geojson(geos))
 
-def tract_centroids_in_json_format(request):
-    censusgeos = get_censustract_geos(request)
+def format_geo_to_geojson(geos):
     # We already have the json strings per model pre-computed, so just place
     # them inside a static response
     response = '{"crs": {"type": "link", "properties": {"href": '
     response += '"http://spatialreference.org/ref/epsg/4326/", "type": '
     response += '"proj4"}}, "type": "FeatureCollection", "features": [%s]}'
-    return response % ', '.join(geo.tract_centroids_as_geojson() for geo in censusgeos)
+    return response % ', '.join(geo.tract_centroids_as_geojson() for geo in geos)
 
-def get_censustract_geoids(request):
-    if 'neLat' in request.GET:
-        geos = get_censustract_geos(request)
-    else:
-        MSA = Geo.objects.get(geo_type=Geo.METRO_TYPE, geoid=request.GET.get('metro'))
-        geos = MSA.get_censustract_geos_by_msa()
-    return geos.values_list('geoid', flat=True)
+def get_censustract_geos(request):
+    northEastLat = request.GET.get('neLat')
+    northEastLon = request.GET.get('neLon')
+    southWestLat = request.GET.get('swLat')
+    southWestLon = request.GET.get('swLon')
+    metro = request.GET.get('metro')
+    if northEastLat and northEastLon and southWestLat and southWestLon:
+        maxlat, minlon, minlat, maxlon = float(northEastLat), float(southWestLon), float(southWestLat), float(northEastLon)
+        geos = get_geos_by_bounds_and_type(maxlat, minlon, minlat, maxlon)
+    elif metro:
+        msa = Geo.objects.get(geo_type=Geo.METRO_TYPE, geoid=metro)
+        geos = msa.get_censustract_geos_by_msa()
+    return geos
```

Comment:
Again, this looks like a function that should have tests.  It'd be very easy to mix up the order of these four fields and not notice it.